sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -146,73 +143,89 @@ specified explicitly.
 ====
 
+Change to SYCL 2020 section `5.9.2 Common address space deduction rules`:
+Namespace scope: if the variable is `work_group_static` object,
+then the variable is assigned to the local address space.
+Otherwise normal rules applies.
+
 SYCL 2020 requires that all global variables accessed by a device function are
 `const` or `constexpr`. This extension lifts that restriction for
 `work_group_static` variables.
 
-[NOTE]
-====
-Since `work_group_static` acts as a view, wrapping an underlying pointer, a
-developer may still choose to declare variables as `const`.
-====
-
 When `T` is a class type or bounded array, the size of the allocation is known
-at compile-time, and a SYCL implementation may embed the size of the allocation
+at compile-time, and a SYCL implementation embeds the size of the allocation
 directly within a kernel. Each instance of `work_group_static<T>` is associated
 with a unique allocation in device local memory.
 
-When `T` is an unbounded array, the size of the allocation is unknown at
-compile-time, and must be communicated to the SYCL implementation via the
-`work_group_static_memory_size` property. Every instance of `work_group_static`
-for which `T` is an unbounded array is associated with a single, shared,
-allocation in device local memory. For example, two instances declared
-as `work_group_static<int[]>` and `work_group_static<float[]>` will be
-associated with the same shared allocation.
-
-If the total amount of device local memory requested (i.e., the sum of
-all memory requested by `local_accessor`, `group_local_memory`,
-`group_local_memory_for_overwrite` and `work_group_static`) exceeds a device's
-local memory capacity (as reported by `local_mem_size`) then the implementation
-must throw a synchronous `exception` with the `errc::memory_allocation` error
-code from the kernel invocation command (e.g. `parallel_for`).
-
 [source,c++]
 ----
-operator T&() const noexcept;
+operator T&() noexcept;
 ----
 _Returns_: A reference to the object stored in the device local memory
 associated with this instance of `work_group_static`.
 
 [source,c++]
 ----
-const work_group_static<T>& operator=(const T& value) const noexcept;
+work_group_static<T>& operator=(const T& value) noexcept;
 ----
-_Constraints_: Available only if `std::is_array_v<T>>` is false.
+_Constraints_: Available only if `std::is_array_v<T>` is false.
 
 _Effects_: Replaces the value referenced by `*ptr` with `value`.
 
 _Returns_: A reference to this instance of `work_group_static`.
 
 [source,c++]
 ----
-T* operator&() const noexcept;
+T* operator&() noexcept;
 ----
 _Returns_: A pointer to the device local memory associated with this
 instance of `work_group_static` (i.e., `ptr`).
 
+=== `get_dynamic_work_group_memory` function
+
+The `get_dynamic_work_group_memory` function provides access
+to a dynamically allocated buffer in the device local memory.
+
+[source,c++]
+----
+template <typename T>
```

Comment:
Why is a get function templated on `T`? When reading the code this suggests that the pointer depends on `T`. I think `static_cast<float*>(work_group_scratch_memory())` or `cast_work_group_scratch_memory<float*>()` would be less confusing. Either makes it clear that this isn't getting a T specific pointer but only casts the single one to T. Note this is independent on whether we use dynamic or scratch as name.